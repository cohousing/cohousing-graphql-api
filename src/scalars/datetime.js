import {GraphQLScalarType} from 'graphql';
import {Kind} from 'graphql/language';
import fs from 'fs';
import moment from 'moment';

export const DateTimeSchema = fs.readFileSync(__dirname + '/datetime.graphqls', 'UTF-8');

export const DateTimeResolver = {
    DateTime: new GraphQLScalarType({
        name: 'DateTime',
        description: 'DateTime type represented in ISO (YYYY-MM-DD HH:mm:ss',
        parseValue(value) {
            console.log('parseValue', value);
            return new Date(value); // value from the client
        },
        serialize(value) {
            return moment(value).format('YYYY-MM-DD HH:mm:ss'); // value sent to the client
        },
        parseLiteral(ast) {
            console.log('parseLiteral', ast);
            if (ast.kind === Kind.INT) {
                return parseInt(ast.value, 10); // ast value is always in string format
            }
            return null;
        }
    })
};