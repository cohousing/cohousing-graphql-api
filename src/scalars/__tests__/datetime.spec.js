import {DateTimeSchema, DateTimeResolver} from '../datetime';

it('should work', () => {
    expect(DateTimeSchema).toBe('scalar DateTime');
});