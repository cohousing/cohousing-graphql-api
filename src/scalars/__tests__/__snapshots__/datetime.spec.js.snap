// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime schema should match 1`] = `"scalar DateTime"`;
